### PR TITLE
DAOS-14981 gurt: restore d_getenv_int undefined symbol

### PR DESCRIPTION
**Pull request by knard-intel**
### Description

Restore missing plain function d_getenv_int() to fix missing symbol with libdaos.

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/include/gurt/common.h [hunk 1]
```diff
@@ -586,6 +586,9 @@ int
 d_getenv_bool(const char *name, bool *bool_val);
 int
 d_getenv_char(const char *name, char *char_val);
+/* DAOS-14981 XXX d_getenv_int() is deprecated, please use d_getenv_uint() */
```
Comment (ashleypittman) [inline src/include/gurt/common.h:589]: There's a compiler intrinsic you should use here instead which will make the compiler warn if the function is used.

File: src/include/gurt/common.h [hunk 2]
```diff
@@ -586,6 +586,9 @@ int
...
+int
+d_getenv_int(const char *name, unsigned int *uint_val);
```
Comment (ashleypittman) [inline src/include/gurt/common.h:591]: I'm not at my desk so can't try it but this should work.
```suggestion
int
d_getenv_int(const char *name, unsigned int *uint_val) __attribute__ ((deprecated(use d_getenv_uint));
```

Comment (knard-intel) [reply src/include/gurt/common.h:591]: - [ ] Use compiler intrinsic macro.

Comment (knard-intel) [reply src/include/gurt/common.h]: > * [x]  Use compiler intrinsic macro.

Fixed with commit e95531c0de297d42cdb888cb34db66305a34003c

**Review by frostedcmos — APPROVED** [review]

**Review by ashleypittman — APPROVED** [review]

**Comment by knard-intel** [comment]
CI failures are known issues:
- CI issue [daos_test/suite.py:DaosCoreTest.test_daos_oid_allocator](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13622/4/testReport/FTEST_daos_test/DaosCoreTest/Test_Hardware___Functional_Hardware_Medium_Verbs_Provider___21___daos_test_suite_py_DaosCoreTest_test_daos_oid_allocator_run_agent_config_transport_config_daos_tests_args_crt_timeout_daos_test_num_clients_pools_created_scalable_endpoint_stopped_ranks_test_name_dmg_hosts_pool_server_config_engines_0_storage_0_1_timeouts_abd8): JIRA ticket [DAOS-14969 ](https://daosio.atlassian.net/browse/DAOS-14969)
- CI issue [pool/create.py:PoolCreateTests.test_create_no_space_loop](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13622/4/testReport/FTEST_pool/PoolCreateTests/Test_Hardware___Functional_Hardware_Medium___1___pool_create_py_PoolCreateTests_test_create_no_space_loop_run_hosts_pool_1_pool_2_server_config_engines_0_storage_0_1_timeouts_67c9/):  JIRA ticket [DAOS-14884](https://daosio.atlassian.net/browse/DAOS-14884)

**Review by gnailzenh — APPROVED** [review]

**Pull request merged**